torchrl/envs/model_based.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from __future__ import annotations
+
+import abc
+from typing import Optional, Union, List
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from torchrl.data import TensorDict
+from torchrl.modules.tensordict_module.world_models import WorldModelWrapper
+from ..data.utils import DEVICE_TYPING
+from ..modules.tensordict_module import TensorDictModule, TensorDictSequence
+from .common import EnvBase
+
+
+class ModelBasedEnv(EnvBase, metaclass=abc.ABCMeta):
+    """
+    Basic environnement for Model Based RL algorithms.
+
+    This class is a wrapper around the model of the MBRL algorithm.
+    This class is meant to give an env framework to a world model and a reward model.
```

Comment:
do we want to cover done too?
also we could cover other things (safety constraints etc).
Perhaps something like:
`This class is meant to give an env framework to a world model (including but not limited to observations, reward, done state and safety constraints models).`